modules/xfem/src/efa/EFAFragment2D.C

```diff
@@ -334,355 +334,69 @@ EFAFragment2D::split()
   // new fragments and return them.
   // N.B. each boundary each can only have 1 cut at most
   std::vector<EFAFragment2D *> new_fragments;
-  std::vector<unsigned int> cut_edges;
-  std::vector<unsigned int> cut_nodes;
+  std::vector<std::vector<EFANode *>>
+      fragment_nodes;           // vector of vectors of EFA nodes for each new fragment
+  unsigned int frag_number = 0; // fragment number in vector of vectors either 0 or 1
   for (unsigned int iedge = 0; iedge < _boundary_edges.size(); ++iedge)
   {
-    if (_boundary_edges[iedge]->getNode(0)->category() == EFANode::N_CATEGORY_EMBEDDED_PERMANENT)
-      cut_nodes.push_back(iedge);
-    if (_boundary_edges[iedge]->numEmbeddedNodes() > 1)
-      EFAError("A fragment boundary edge can't have more than 1 cuts");
-    if (_boundary_edges[iedge]->hasIntersection())
-      cut_edges.push_back(iedge);
-  }
 
-  if (cut_edges.size() > 2)
-  {
-    EFAError("In split() fragment cannot have more than 2 cut edges");
-  }
-  else if (cut_edges.size() == 1 && cut_nodes.size() == 1)
```

Comment:
This part of code was to deal with the cutting through nodes cases. Do we still have that capability?

Reply:
He says that he handles that. I need to look through the code carefully and convince myself about that. The tests supposedly pass, which is a good sign.